### PR TITLE
Builder pattern in Task and merged attachment as note into Task only

### LINKED ISSUE #7: What happens when no attachment(note) is added to the task while creating task reminder?
The `Task` class refers to `long attachmentId` as a foreign key from the `Attachments` table. How can we store a task object in the database which does not have a note?
### Problem
If we store a zero, then these tasks will start pointing to the attachement with `id = 0`. 
### Possible solutions
1. One possible solution would be to place a dummy null attachment at 0 id, which can be referred and checked.
2. Should we merge the `attachmentContent` column to the `Tasks` table only? (Preferred)

### PR DESCRIPTION
**Pull request by vermayash8**
Resolved #7 by making note as a part of `Task` class.

### CODE REVIEW

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Task.java [hunk 1]
```diff
@@ -280,7 +292,157 @@ public Date getDateAdded() {
...
+     */
+    public static class Builder {
+
+        private String taskName;
+        private long locationId;
+        // Assigning default values to the remaining.
+        private String imageUri = null;
+        private int isDone = 0;
+        private int isAlarmSet = 1;
+        private int reminderRange = 75;
```
Comment (Shilpi75) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Task.java]: This should be from default settings

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Task.java [hunk 2]
```diff
@@ -280,7 +292,157 @@ public Date getDateAdded() {
...
+         * by the setNextStartDate() method. Therefore, we're initially setting it as null so
+         * that we can set it equal to startDate if it's still null at the time of building.
+         */
+        private Date nextStartDate = null;
+        private int repeatType = DbConstants.NO_REPEAT;
+        private int movementType = DbConstants.BOTH_ENTER_EXIT;
+        private int activityType = DbConstants.ACTIVITY_ANYTHING;
+        private float lastDistance = Integer.MAX_VALUE;
+        private Date lastTriggered = null;
+        private long snoozedAt = 0L;
```
Comment (Shilpi75) [inline app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Task.java]: Sure about this value?

Comment (vermayash8) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Task.java]: Yes, `0` or `-1` in the snoozed time will tell us that the alarm wasn't snoozed at all. We're sure that 0 won't come anytime in the actual case.

Comment (Shilpi75) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Task.java]: What about keeping it as -1?

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Task.java (hunk 1 above)
Comment (vermayash8) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Task.java]: Resolved.

File: app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Task.java (hunk 2 above)
Comment (vermayash8) [reply app/src/main/java/app/tasknearby/yashcreations/com/tasknearby/models/Task.java]: Placed `-1`.

**Pull request merged**